### PR TITLE
Added mechanism to set human readable name while import cluster flow

### PR DESCRIPTION
**Pull request by shtripat**
tendrl-bug-id: Tendrl/commons#888
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

**Comment by shtripat** [comment]
THIS IS WIP. YET TO VERIFY WITH ALL THE SCENARIOS.

**Comment by shtripat** [comment]
Verified the changes with https://github.com/Tendrl/monitoring-integration/pull/394 with below scenarios
- Import cluster without passing `Cluster.short_name` field
- Import cluster with `Cluster.short_name` field
- Un-manage a cluster imported with short_name

Works as expected. API module doesnt need any changes for this and UI need changes to pass `Cluster.short_name` (optional) if human readable name to be used for cluster. Also UI should display the short_name if available for a cluster.

@gnehapk ^^^

File: tendrl/commons/flows/import_cluster/__init__.py [hunk 1]
```diff
@@ -13,6 +14,28 @@ def __init__(self, *args, **kwargs):
...
+                    )
+            # Check for uniqueness of cluster short name
+            _clusters = NS._int.client.read(
+                '/clusters'
+            )
+            for entry in _clusters.leaves:
+                _cluster = NS.tendrl.objects.Cluster(
+                    integration_id=entry.key.split('/')[-1]
+                ).load()
+                if _cluster.short_name and short_name and \
```
Comment (nthomas-redhat) [inline tendrl/commons/flows/import_cluster/__init__.py]: covert to lower case and check? remove the leading space if any as well

File: tendrl/commons/objects/cluster/atoms/setup_cluster_alias/__init__.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            payload=payload
+        ).save()
+
+        # Wait for 2 mins for the job to complete
+        loop_count = 0
+        wait_count = 24
+        while True:
+            if loop_count >= wait_count:
+                logger.log(
+                    "info",
```
Comment (nthomas-redhat) [inline tendrl/commons/objects/cluster/atoms/setup_cluster_alias/__init__.py]: Why the log level is info? its a failure scenario right?

File: tendrl/commons/flows/import_cluster/__init__.py (hunk 1 above)
Comment (shtripat) [reply tendrl/commons/flows/import_cluster/__init__.py]: done

File: tendrl/commons/objects/cluster/atoms/setup_cluster_alias/__init__.py (hunk 2 above)
Comment (shtripat) [reply tendrl/commons/objects/cluster/atoms/setup_cluster_alias/__init__.py]: This is a job update and should be info type

File: tendrl/commons/flows/import_cluster/__init__.py [hunk 3]
```diff
@@ -13,6 +14,28 @@ def __init__(self, *args, **kwargs):
     def run(self):
         if "Node[]" not in self.parameters:
             integration_id = self.parameters['TendrlContext.integration_id']
+            short_name = self.parameters.get('Cluster.short_name', None)
```
Comment (r0h4n) [inline tendrl/commons/flows/import_cluster/__init__.py]: please add check for max 64 characters.

File: tendrl/commons/objects/cluster/atoms/setup_cluster_alias/__init__.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        _job_id = str(uuid.uuid4())
+        payload = {
+            "tags": ["tendrl/integration/monitoring"],
+            "run": "monitoring.flows.SetupClusterAlias",
+            "status": "new",
+            "parameters": self.parameters,
+            "type": "monitoring"
+        }
+        Job(
+            job_id=_job_id,
```
Comment (r0h4n) [inline tendrl/commons/objects/cluster/atoms/setup_cluster_alias/__init__.py]: add parent job id

**Pull request merged**